haskell-decl-scan.el

```diff
@@ -382,18 +377,17 @@ there."
     nil))
 
 (defun haskell-ds-line-commented-p ()
-  "Tests if all characters from `point' to `end-of-line' pass
-`haskell-ds-comment-p'"
-  (let ((r t))
-    (while (and r (not (eolp)))
-      (if (not (haskell-ds-comment-p))
-          (setq r nil))
-      (forward-char))
-    r))
+  "Test if all characters from `point' to `end-of-line' pass `haskell-ds-comment-p'."
```

Comment:
The only usage of this function is to go back to the end of a definition.  That requires going through all lines that are composed only of white space and comments.